## NGS data analysis tools

This repository contains command line tools for analysis of next-generation sequencing (NGS) data in `.fastq`, `.sam` and `.bam` file formats generated by single or paired-end sequencing, such as ChIP-seq, ATAC-seq and RNA-seq. 

The Shell scripts include the following analyses: 

 - NGS data pre-processing (`phantom`, `picard`) 
 - Peak calling (`MACS2`) 
 - Signal track generaton (`align2rawsignal`, `picard`) 
 - Exploratory analysis (`picard`, `deepTools`)
 - Heatmap and TSS profile generation for chromatin binding and accessiblity (`deepTools`) 
 - Differential motif enrichment analysis (`HOMER`) 
 - Reference genome generation (e.g. `hg38`, `mm10`) 

### Documentation and References:

https://www.encodeproject.org/software/phantompeakqualtools/

https://github.com/kundajelab/phantompeakqualtools

https://broadinstitute.github.io/picard/

https://pypi.org/project/MACS2/

https://github.com/akundaje/align2rawsignal

https://code.google.com/archive/p/align2rawsignal/

https://deeptools.readthedocs.io/en/develop/index.html

http://homer.ucsd.edu/homer/motif/

The scripts `run_spp.R` and `run_spp_nudups.R` called in `phantom.sh` are by:

[1] Anshul Kundaje, Computer Science Dept., MIT, ENCODE Consortium, http://code.google.com/p/phantompeakqualtools, Feb 2013

[2] Kharchenko PK, Tolstorukov MY, Park PJ, Design and analysis of ChIP-seq experiments for DNA-binding proteins Nat Biotechnol. 2008

Please see README.txt for more information.


